private/byte.rkt

```diff
@@ -109,3 +116,279 @@
+
+      (check-equal? (byte-ref (byte-nor x (byte 0 1 0 0 0 0 0 0)) 1) 0)
+      (check-equal? (byte-ref (byte-nor (byte 0 1 0 0 0 0 0 0) x) 1) 0)
+
+      (check-equal? (byte-ref (byte-nor x (byte 0 0 1 0 0 0 0 0)) 2) 0)
+      (check-equal? (byte-ref (byte-nor (byte 0 0 1 0 0 0 0 0) x) 2) 0)
+
+      (check-equal? (byte-ref (byte-nor x (byte 0 0 0 1 0 0 0 0)) 3) 0)
+      (check-equal? (byte-ref (byte-nor (byte 0 0 0 1 0 0 0 0) x) 3) 0)
+
+      (check-equal? (byte-ref (byte-nor x (byte 0 0 0 0 1 0 0 0)) 4) 0)
+      (check-equal? (byte-ref (byte-nor (byte 0 0 0 0 1 0 0 0) x) 4) 0)
+
+      (check-equal? (byte-ref (byte-nor x (byte 0 0 0 0 0 1 0 0)) 5) 0)
+      (check-equal? (byte-ref (byte-nor (byte 0 0 0 0 0 1 0 0) x) 5) 0)
+      
+      (check-equal? (byte-ref (byte-nor x (byte 0 0 0 0 0 0 1 0)) 6) 0)
+      (check-equal? (byte-ref (byte-nor (byte 0 0 0 0 0 0 1 0) x) 6) 0)
+
+      (check-equal? (byte-ref (byte-nor x (byte 0 0 0 0 0 0 0 1)) 7) 0)
+      (check-equal? (byte-ref (byte-nor (byte 0 0 0 0 0 0 0 1) x) 7) 0))))
+
+(define (byte-xnor left right)
+  (apply byte
+         (for/list ([i (in-range 8)])
+           (bitwise-xor (bitwise-xor (byte-ref left i)
+                                     (byte-ref right i))
+                        1)))) 
+
+(module+ test
+  (test-case
+    "byte-xor"
+    (for ([x (in-range 256)])
+      (check-equal? (byte-ref (byte-xnor x 0) 0)
+                    (byte-ref (byte-not x) 0))
+      (check-equal? (byte-ref (byte-xnor x 255) 0)
+                    (byte-ref x 0))
+
+      (check-equal? (byte-ref (byte-xnor x 0) 1)
+                    (byte-ref (byte-not x) 1))
+      (check-equal? (byte-ref (byte-xnor x 255) 1)
+                    (byte-ref x 1))
+
+      (check-equal? (byte-ref (byte-xnor x 0) 2)
+                    (byte-ref (byte-not x) 2))
+      (check-equal? (byte-ref (byte-xnor x 255) 2)
+                    (byte-ref x 2))
+
+      (check-equal? (byte-ref (byte-xnor x 0) 3)
+                    (byte-ref (byte-not x) 3))
+      (check-equal? (byte-ref (byte-xnor x 255) 3)
+                    (byte-ref x 3))
+
+      (check-equal? (byte-ref (byte-xnor x 0) 4)
+                    (byte-ref (byte-not x) 4))
+      (check-equal? (byte-ref (byte-xnor x 255) 4)
+                    (byte-ref x 4))
+
+      (check-equal? (byte-ref (byte-xnor x 0) 5)
+                    (byte-ref (byte-not x) 5))
+      (check-equal? (byte-ref (byte-xnor x 255) 5)
+                    (byte-ref x 5))
+
+      (check-equal? (byte-ref (byte-xnor x 0) 6)
+                    (byte-ref (byte-not x) 6))
+      (check-equal? (byte-ref (byte-xnor x 255) 6)
+                    (byte-ref x 6))
+
+      (check-equal? (byte-ref (byte-xnor x 0) 7)
+                    (byte-ref (byte-not x) 7))
+      (check-equal? (byte-ref (byte-xnor x 255) 7)
+                    (byte-ref x 7)))))
+                    
+(define (in-byte b)
+  (make-do-sequence
```

Comment:
The `make-do-sequence` function is super complicated and hard to use. If you like, instead you could return the list directly because lists implement the sequence interface.

```
(define (in-byte b)
  (list (byte-ref b 0) (byte-ref b 1) ...))
```